compress.cc

```diff
@@ -97,6 +97,10 @@ std::map<sstring, sstring> compressor::options() const {
     return {};
```

Comment:
Needs unit tests / perf tests (how much of memory footprint do we save?)

Reply:
Seems that should have been made for zstd, when all hallaballo regarding sharings its internal structures came ito being. This patch simply generalizes sharing across all compressors.